k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -0,0 +1,241 @@
+      edm4hep::ConstParticleID pIDUsed = edm_rp.getParticleIDUsed();
+      if (pIDUsed.isAvailable()) {
+        for (auto& particleID : lcio_particleIDs_vec) {
+          if (particleID.second == pIDUsed) {
+            lcio_recp->setParticleIDUsed(particleID.first);
+            break;
+          }
+        }
+      }
+    }
+
+    // Associated Vertex
+    edm4hep::ConstVertex vertex = edm_rp.getStartVertex();
+    if (vertex.isAvailable()) {
+      auto* lcio_vertex = new lcio::VertexImpl;
+      lcio_vertex->setPrimary( vertex.getPrimary() );
+      #warning "AlgoritymType conversion from int to string"
+      lcio_vertex->setAlgorithmType( std::string{vertex.getAlgorithmType()} ); // TODO std::string(int)
+      lcio_vertex->setChi2( vertex.getChi2() );
+      lcio_vertex->setProbability( vertex.getProbability() );
+      lcio_vertex->setPosition( vertex.getPosition()[0], vertex.getPosition()[1], vertex.getPosition()[2]  );
+      lcio_vertex->setCovMatrix( vertex.getCovMatrix().data() );
+      lcio_recp->setStartVertex(lcio_vertex);
+
+      // Associated particle to the vertex
+      edm4hep::ConstReconstructedParticle vertex_rp = vertex.getAssociatedParticle();
+      if (vertex_rp.isAvailable()) {
+        // TODO add rp
+      }
+    }
+
+    // Tracks
+    for (int j=0; j < edm_rp.tracks_size(); ++j) {
+      edm4hep::ConstTrack edm_rp_tr = edm_rp.getTracks(j);
+      for (auto& track : lcio_tracks_vec) {
+        if (track.second == edm_rp_tr) {
+          lcio_recp->addTrack(track.first);
+        }
+      }
+    }
+
+    // Add converted LCIO RecoParticle, and associated EDM4hep RecoParticle
+    lcio_rec_particles_vec.emplace_back(
+      std::make_pair(lcio_recp, edm_rp)
+    );
+  }
+}
+
+
+void EDM4hep2LcioTool::convert(
+  const std::string& type,
+  const std::string& name)
+{
+  if (type == "edm4hep::TrackCollection") {
+    addLCIOConvertedTracks(lcio_tracks_vec, name);
+  } else
+  if (type == "edm4hep::ParticleIDCollection") {
+    addLCIOParticleIDs(lcio_particleIDs_vec, name);
+  } else
+  if (type == "edm4hep::ReconstructedParticleCollection") {
+    addLCIOReconstructedParticles(
+      lcio_rec_particles_vec,
+      lcio_particleIDs_vec,
+      lcio_tracks_vec,
+      name);
+  } else {
+    error() << "Error trying to convert requested " << type << " with name " << name << endmsg;
+  }
+}
+
+
+StatusCode EDM4hep2LcioTool::convertCollections(
+  const Gaudi::Property<std::vector<std::string>>& parameters,
+  lcio::LCEventImpl* lcio_event)
+{
+  info() << "Converting EDM4hep to LCIO requested collections. " << endmsg;
+
+  if (parameters.size() % 3 != 0) {
+    error() << " Error processing conversion parameters. 3 arguments per collection expected. " << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  for (int i = 0; i < parameters.size(); i=i+3) {
+    convert(parameters[i], parameters[i+1]);
+  }
+
+  // Populate LCIO event with EDM4hep info
+  auto* tracks = new lcio::LCCollectionVec(lcio::LCIO::TRACK);
```

Comment:
Need to loop over the collections to convert, and then create those (unless they already exist in the LCIO event)